File: backend/app/schemas/pricing.py
"""Pricing and Financial Breakdown Schemas."""

from decimal import Decimal
from typing import Optional
from pydantic import BaseModel, Field


class DeliveryPricingResponse(BaseModel):
    delivery_id: str
    base_fare: Decimal = Field(..., description="Base mobilization fare.")
    distance_km: float = Field(..., description="Calculated road travel distance in kilometers.")
    distance_charge: Decimal = Field(..., description="Distance component of delivery fee.")
    duration_minutes: int = Field(..., description="Estimated travel duration in minutes.")
    time_charge: Decimal = Field(..., description="Time component of delivery fee.")
    stop_count: int = Field(..., description="Number of delivery stops.")
    stop_fees: Decimal = Field(..., description="Aggregated stop dropoff fees.")
    delivery_charge: Decimal = Field(..., description="Subtotal delivery charge.")
    platform_fee_percent: Decimal = Field(..., description="Configured platform/service fee percentage (default 12%).")
    platform_fee: Decimal = Field(..., description="12% platform/service fee.")
    ngo_total: Decimal = Field(..., description="Total amount payable by receiving NGO (delivery charge + platform fee).")
    driver_payout: Decimal = Field(..., description="Calculated driver compensation.")
    currency: str = "INR"

File: anna-setu-frontend-development/lib/api/pricing.ts
/**
 * AnnaSetu — Delivery Pricing & Fee Breakdown API Client
 */

import { apiClient } from './client'

export interface DeliveryPricing {
  delivery_id: string
  base_fare: number
  distance_km: number
  distance_charge: number
  duration_minutes: number
  time_charge: number
  stop_count: number
  stop_fees: number
  delivery_charge: number
  platform_fee_percent: number
  platform_fee: number
  ngo_total: number
  driver_payout: number
  currency: 'INR'
}

export const pricingApi = {
  /**
   * Calculate exact pricing breakdown for a delivery mission
   */
  async getDeliveryPricing(deliveryId: string): Promise<DeliveryPricing> {
    return apiClient<DeliveryPricing>(`/api/v1/pricing/delivery/${deliveryId}`)
  },
}
